src/main/java/org/spongepowered/api/Platform.java

```diff
@@ -24,21 +24,74 @@
  */
 package org.spongepowered.api;
 
+import java.util.Map;
+
 /**
- * Effective side platforms.
- *
- * <p>A side is what part of Minecraft this is being run on. The client, or the
- * server. The internal server is also treated like a dedicated server.</p>
+ * Represents a possible platform, or implementation, a {@link Game} could be running on.
  */
-public enum Platform {
+public interface Platform {
+
+    /**
+     * Retrieves the current {@link Side} this platform is running on.
+     *
+     * @return The current side
+     */
+    Side getSide();
+
+    /**
+     * Retrieves the current platform name.
+     *
+     * @return The platform name
+     */
+    String getName();
+
+    /**
+     * Retrieves the current platform version.
+     *
+     * <p>This version can be in any format, like build number of semver.</p>
+     *
+     * @return The platform version, as a string
+     */
+    String getVersion();
+
+    /**
+     * Retrieves the current Sponge API version that this platform is implementing.
+     *
+     * @return The API version
+     */
+    String getApiVersion();
+
+    /**
+     * Gets current Minecraft version of this platform.
+     *
+     * @return The Minecraft version
+     */
+    MinecraftVersion getMinecraftVersion();
 
     /**
-     * The platform of a Minecraft CLIENT is expected.
+     * Returns this platform instance, as a key-value map.
+     *
+     * @return This playform as a map
      */
-    CLIENT,
+    Map<String, ?> asMap();
```

Comment:
That does make sense, and could be useful, I agree. I amended the javadocs to `#asMap` to say that modifying the map directly modifies the platform instance. This could be implemented as a proxy to a hash map, where keys that exist are set to fields directly.